### PR TITLE
EAR-1473-piped-meta-not-displaying-as-deleted

### PR DESCRIPTION
**Pull request by zimboflyman**
### What is the context of this PR?

When piping in a metadata value into a question Title and then deleting the relevant row from the metadata table, it does not display the placeholder [Deleted metadata] in the left navigation bar. 

Previously, [deleted metadata] in the question title was being handled in the front end and not updating the cache
This Pr seeks to move this process to the resolver so that the cache is updated and therefor the navigation sidebar is reRendered 😀

### How to review

1. Create a new Question in your Questionnaire
2. add in a piped metadata in the question title
3. go to metadata page and delete that metadata
4. Both the Question title _**and**_ side navbar title for that question should ready [Deleted metadata]

also test for multiple questions with piped metadata in the title when deleting the relevant metadata

Also check the Question description is updated to include [deleted metadata] if it has metadata piped that has been deleted

UNSURE IF THIS WOULD NEED A MIGRATION  ???

### CODE REVIEW

**Review by jasehumphr — DISMISSED** [review]
Migration looks Sweet

File: eq-author-api/migrations/upateAnswerDeletedMetaData.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    if (page?.title?.includes("metadata")) {
+      const splitTitleToArray = page.title.split(/(?<=\/span)/);
+
+      splitTitleToArray.map((titlePart) => {
+        const idExists = regexp.exec(titlePart);
+
+        if (idExists !== null && titlePart.includes("metadata")) {
+          const index = splitTitleToArray.indexOf(titlePart);
+
+          const titleMetadtaId = idExists[0];
```
Comment (farres1) [inline eq-author-api/migrations/upateAnswerDeletedMetaData.js:21]:
```suggestion
          const titleMetadataId = idExists[0];
```
Dismiss these requested changes if they are not important

File: eq-author-api/migrations/upateAnswerDeletedMetaData.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          if (
+            !metadata.some((thisMetaData) => thisMetaData.id === titleMetadtaId)
```
Comment (farres1) [inline eq-author-api/migrations/upateAnswerDeletedMetaData.js:23]:
```suggestion
            !metadata.some((thisMetaData) => thisMetaData.id === titleMetadataId)
```

File: eq-author-api/migrations/upateAnswerDeletedMetaData.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    if (page?.description?.includes("metadata")) {
+      const splitdescriptionToArray = page.description.split(/(?<=\/span)/);
+
+      splitdescriptionToArray.map((titlePart) => {
+        const idExists = regexp.exec(titlePart);
+
+        if (idExists !== null && titlePart.includes("metadata")) {
+          const index = splitdescriptionToArray.indexOf(titlePart);
+          const titleMetadtaId = idExists[0];
```
Comment (farres1) [inline eq-author-api/migrations/upateAnswerDeletedMetaData.js:41]:
```suggestion
          const titleMetadataId = idExists[0];
```

File: eq-author-api/migrations/upateAnswerDeletedMetaData.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          if (
+            !metadata.some((thisMetaData) => thisMetaData.id === titleMetadtaId)
```
Comment (farres1) [inline eq-author-api/migrations/upateAnswerDeletedMetaData.js:43]:
```suggestion
            !metadata.some((thisMetaData) => thisMetaData.id === titleMetadataId)
```

**Review by farres1 — CHANGES_REQUESTED** [review]
Changes to the name of a value in the migration. Dismiss these requested changes if these are not important

**Review by farres1 — APPROVED** [review]

**Review by jasehumphr — APPROVED** [review]

**Pull request merged**